### PR TITLE
Refactor: Make bundle using webpack

### PR DESCRIPTION
**Pull request by Tainan404**
### What does this PR do?
Implements webpack and remove meteor as bundler

### CODE REVIEW

File: build/packages-template/bbb-html5/opts-jammy.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 . ./opts-global.sh
 
-OPTS="$OPTS -d bc,bbb-pads,bbb-webrtc-sfu,bbb-export-annotations,bbb-web,bbb-html5-nodejs,bbb-graphql-middleware,bbb-graphql-actions,yq,mongodb-org -t deb"
+OPTS="$OPTS -d bc,bbb-pads,bbb-webrtc-sfu,bbb-export-annotations,bbb-web,bbb-graphql-middleware,bbb-graphql-actions,yq,nginx -t deb"
```
Comment (antobinary) [inline build/packages-template/bbb-html5/opts-jammy.sh:3]:
```suggestion
OPTS="$OPTS -d bc,bbb-pads,bbb-webrtc-sfu,bbb-export-annotations,bbb-web,bbb-graphql-middleware,bbb-graphql-actions,yq,nginx,unzip -t deb"
```

File: build/packages-template/bbb-html5/build.sh [hunk 2]
```diff
@@ -12,117 +12,51 @@ BUILD=$1
...
 fpm -s dir -C ./staging -n $PACKAGE \
     --version $VERSION --epoch $EPOCH \
     --before-install before-install.sh \
-    --after-install after-install.sh \
-    --before-remove before-remove.sh \
-    --after-remove after-remove.sh \
-    --description "The HTML5 components for BigBlueButton" \
+    --after-install after-install.sh        \
+    --after-remove  after-remove.sh        \
+    --depends unzip                        \
```
Comment (antobinary) [inline build/packages-template/bbb-html5/build.sh:59]: Let's move `unzip` in `build/packages-template/bbb-html5/opts-jammy.sh`

File: build/packages-template/bbb-html5/before-install.sh [hunk 3]
```diff
@@ -1,28 +1 @@
...
-
-    if [ -f /usr/share/meteor/bundle/programs/server/node_modules ]; then
-      rm -r /usr/share/meteor/bundle/programs/server/node_modules
-    fi
-
-    # Remove remnants from old architecture prior to BBB 3.0.x-alpha.6
-    if [ -f /usr/lib/systemd/system/bbb-html5-backend@.service ]; then
-      rm /usr/lib/systemd/system/bbb-html5-backend@.service
-    fi
-    if [ -f /usr/lib/systemd/system/bbb-html5-frontend@.service ]; then
```
Comment (antobinary) [inline build/packages-template/bbb-html5/before-install.sh:17]: Please keep these cleanup procedures - they were put to handle upgrades from earlier 3.0 BBB installations. We can remove them safely in the future (3.1+)

File: build/packages-template/bbb-html5/bbb-html5-conn-limit.conf [hunk 4]
```diff
@@ -1 +0,0 @@
-limit_conn_zone $uri zone=ws_zone:5m;
```
Comment (antobinary) [inline build/packages-template/bbb-html5/bbb-html5-conn-limit.conf:1]: Is there a similar functionality now? 

> limit of 3 simultaneous connections (by uri) only for location /html5client/sockjs
https://github.com/bigbluebutton/bigbluebutton/pull/16771

File: build/packages-template/bbb-html5/after-install.sh [hunk 5]
```diff
@@ -1,81 +1,10 @@
...
-  || [ -f /usr/lib/systemd/system/mongod.service.d/mongod-service-override.conf ] ; then
-  rm -f /etc/systemd/system/mongod.service.d/override-mongo.conf
-  rm -f /etc/systemd/system/mongod.service.d/override.conf
-  rm -f /usr/lib/systemd/system/mongod.service.d/mongod-service-override.conf 
-  systemctl daemon-reload
-fi
-
-# Enable Listen Only support in FreeSWITCH
-if [ -f /opt/freeswitch/etc/freeswitch/sip_profiles/external.xml ]; then
-  sed -i 's/<!--<param name="enable-3pcc" value="true"\/>-->/<param name="enable-3pcc" value="proxy"\/>/g' /opt/freeswitch/etc/freeswitch/sip_profiles/external.xml
```
Comment (antobinary) [inline build/packages-template/bbb-html5/after-install.sh:70]: @prlanzarin do we still need this in BBB 3.0?

File: build/packages-template/bbb-html5/after-install.sh [hunk 6]
```diff
@@ -1,81 +1,10 @@
...
-  find /usr/share/meteor/bundle/programs/web.browser/app/locales -maxdepth 1 -type f -name "*.json" -exec basename {} \; | awk 'BEGIN{printf "["}{printf "\"%s\", ", $0}END{print "]"}' | sed 's/, ]/]/' > /usr/share/meteor/bundle/programs/web.browser/app/locales/index.json
-fi
-
-# set full BBB version in settings.yml so it can be displayed in the client
-BBB_RELEASE_FILE=/etc/bigbluebutton/bigbluebutton-release
-BBB_HTML5_SETTINGS_FILE=/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml
-if [ -f $BBB_RELEASE_FILE ] && [ -f $BBB_HTML5_SETTINGS_FILE ]; then
-  BBB_FULL_VERSION=$(cat $BBB_RELEASE_FILE | sed -n '/^BIGBLUEBUTTON_RELEASE/{s/.*=//;p}' | tail -n 1)
-  echo "setting public.app.bbbServerVersion: $BBB_FULL_VERSION in $BBB_HTML5_SETTINGS_FILE "
-  yq e -i ".public.app.bbbServerVersion = \"$BBB_FULL_VERSION\"" $BBB_HTML5_SETTINGS_FILE
```
Comment (antobinary) [inline build/packages-template/bbb-html5/after-install.sh:54]: @Tainan404 please link to how you set this field now

File: build/packages-template/bbb-html5/after-install.sh [hunk 7]
```diff
@@ -1,81 +1,10 @@
...
-    chown -R root:root /usr/share/meteor
-fi
-
-TARGET=/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml
-
-  WSURL=$(cat $SERVLET_DIR/WEB-INF/classes/bigbluebutton.properties | grep -v '#' | sed -n '/^bigbluebutton.web.serverURL/{s/.*=//;p}' | sed 's/https/wss/g' | sed s'/http/ws/g')
-
-  yq e -i ".public.kurento.wsUrl = \"$WSURL/bbb-webrtc-sfu\"" $TARGET
-
-  yq e -i  ".public.pads.url = \"$PROTOCOL://$HOST/pad\"" $TARGET
```
Comment (antobinary) [inline build/packages-template/bbb-html5/after-install.sh:30]: I believe we need to keep this or implement it elsewhere

File: build/packages-template/bbb-html5/after-install.sh [hunk 8]
```diff
@@ -1,81 +1,10 @@
 #!/bin/bash -e
 
-
-HOST=$(cat $SERVLET_DIR/WEB-INF/classes/bigbluebutton.properties | grep -v '#' | sed -n '/^bigbluebutton.web.serverURL/{s/.*\///;p}')
-
-if [ ! -L /etc/nginx/sites-enabled/bigbluebutton ]; then
-  mkdir -p /etc/nginx/sites-enabled
-  ln -s /etc/nginx/sites-available/bigbluebutton /etc/nginx/sites-enabled/bigbluebutton
```
Comment (antobinary) [inline build/packages-template/bbb-html5/after-install.sh:8]: I believe we need to keep this or implement it elsewhere

File: build/packages-template/bbb-html5/mongod.service [hunk 9]
```diff
@@ -1,39 +0,0 @@
-#
-# Override file for mongod.service
-#
-[Unit]
-Description=High-performance, schema-free document-oriented database
```
Comment (antobinary) [inline build/packages-template/bbb-html5/mongod.service:5]: Reminder to self to update `bbb-install.sh` for 3.0 to no longer install MongoDB.
Also need to double check upgrading from 3.0.0-alpha.7 to ensure all mongo packages are now listed as not needed.

File: build/packages-template/bbb-html5/mongod.service [hunk 10]
```diff
@@ -1,39 +0,0 @@
...
-After=network.target
-Documentation=https://docs.mongodb.org/manual
-PartOf=bigbluebutton.target
-
```
Comment (antobinary) [inline build/packages-template/bbb-html5/mongod.service:9]: A few more mentions of mongo are found in https://github.com/bigbluebutton/bigbluebutton/blob/v3.0.x-release/bigbluebutton-config/bin/bbb-conf
Let's remove them.

File: bigbluebutton-html5/imports/api/audio/client/bridge/sfu-audio-bridge.js [hunk 11]
```diff
@@ -509,4 +509,4 @@ export default class SFUAudioBridge extends BaseAudioBridge {
   }
 }
 
-module.exports = SFUAudioBridge;
+// module.exports = SFUAudioBridge;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sfu-audio-bridge.js:512]: Could you please comment (here or inline) why we're disabling the export.

File: bigbluebutton-html5/imports/api/audio/client/bridge/sip.js [hunk 12]
```diff
@@ -1307,4 +1307,4 @@ export default class SIPBridge extends BaseAudioBridge {
   }
 }
 
-module.exports = SIPBridge;
+// module.exports = SIPBridge;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/audio/client/bridge/sip.js:1310]: Could you please comment (here or inline) why we're disabling the export.

File: bigbluebutton-html5/package.json [hunk 13]
```diff
@@ -58,7 +69,6 @@
     "eventemitter2": "~6.4.6",
     "fast-json-patch": "^3.1.1",
     "fastdom": "^1.0.10",
-    "fibers": "^4.0.2",
```
Comment (antobinary) [inline bigbluebutton-html5/package.json:61]: Do we still need `"meteor-node-stubs": "^1.2.1",`

File: bigbluebutton-html5/package.json [hunk 14]
```diff
@@ -112,25 +122,50 @@
...
   },
   "devDependencies": {
+    "@babel/core": "7.24.9",
+    "@babel/plugin-transform-runtime": "^7.24.7",
+    "@babel/polyfill": "^7.12.1",
+    "@babel/preset-env": "7.24.8",
+    "@babel/preset-react": "7.24.7",
+    "@babel/preset-typescript": "7.24.7",
     "@types/emoji-mart": "^3.0.9",
     "@types/meteor": "^2.9.3",
```
Comment (antobinary) [inline bigbluebutton-html5/package.json:132]: Do we still need `"@types/meteor": "^2.9.3",`?

File: build/packages-template/bbb-html5/build.sh [hunk 15]
```diff
@@ -12,117 +12,51 @@ BUILD=$1
...
+    --description "BigBlueButton bbb-html5-client" \
```
Comment (antobinary) [inline build/packages-template/bbb-html5/build.sh:60]: Do we actually introduce `bbb-html5-client`? Please consider something more descriptive.

File: build/packages-template/bbb-learning-dashboard/build.sh [hunk 16]
```diff
@@ -23,7 +23,7 @@ done
 
 mkdir -p staging/var/bigbluebutton/learning-dashboard
 
-mkdir -p staging/usr/share/bigbluebutton/nginx
+mkdir -p staging/usr/share/bigbluebutton/
```
Comment (gustavotrott) [inline build/packages-template/bbb-learning-dashboard/build.sh:26]: was that on purpose?

File: build/packages-template/bbb-html5/bbb-html5-conn-limit.conf (hunk 4 above)
Comment (Tainan404) [reply build/packages-template/bbb-html5/bbb-html5-conn-limit.conf:1]: I think the graphql middleware theres a limitation per user, @gustavotrott can confirm it

Comment (gustavotrott) [reply build/packages-template/bbb-html5/bbb-html5-conn-limit.conf:1]: There is no limit, but I will take note to add it for Graphql ws connections!

File: bigbluebutton-html5/imports/api/audio/client/bridge/sfu-audio-bridge.js (hunk 11 above)
Comment (Tainan404) [reply bigbluebutton-html5/imports/api/audio/client/bridge/sfu-audio-bridge.js:512]: we mainly use esm export/import, so this kind of import isn't need anymore and could cause conflicts on compilation

**Comment by antobinary** [comment]
TODO:
- [ ] revisit docs `#### Modify localization for the client (override with the latest locale)` to test if my edits worked.
- [ ] Update the architecture diagram to remove mongo and bbb-html5
- [ ] `isMeteorConnected` can likely be removed
- [ ] docs - drop `#### Meteor / HTML5 Frontend (experimental)`
- [ ] run through the install and dev-guide docs for 3.0. I've edited those but they'd benefit from a test.

CLOSES:
Closes https://github.com/bigbluebutton/bigbluebutton/issues/11609
Closes https://github.com/bigbluebutton/bigbluebutton/issues/15237

**Comment by ramonlsouza** [comment]
closes #16953

**Pull request merged**